examples/bot-ready-signalling/client/javascript/src/app.js

```diff
@@ -0,0 +1,216 @@
+   */
+  log(message) {
+    const entry = document.createElement('div');
+    entry.textContent = `${new Date().toISOString()} - ${message}`;
+
+    // Add styling based on message type
+    if (message.startsWith('User: ')) {
+      entry.style.color = '#2196F3'; // blue for user
+    } else if (message.startsWith('Bot: ')) {
+      entry.style.color = '#4CAF50'; // green for bot
+    }
+
+    this.debugLog.appendChild(entry);
+    this.debugLog.scrollTop = this.debugLog.scrollHeight;
+    console.log(message);
+  }
+
+  /**
+   * Update the connection status display
+   */
+  updateStatus(status) {
+    this.statusSpan.textContent = status;
+    this.log(`Status: ${status}`);
+  }
+
+  handleEventToConsole (evt) {
+    this.log(`Received event: ${evt.action}`);
+  };
+
+  /**
+   * Set up listeners for track events (start/stop)
+   * This handles new tracks being added during the session
+   */
+  setupTrackListeners() {
+    if (!this.dailyCallObject) return;
+
+    this.dailyCallObject.on("joined-meeting", () => {
+      this.updateStatus('Connected');
+      this.connectBtn.disabled = true;
+      this.disconnectBtn.disabled = false;
+      this.log('Client connected');
+    });
+    this.dailyCallObject.on("track-started", (evt) => {
+      if (evt.track.kind === "audio" && evt.participant.local === false) {
+        this.log("Audio track started.")
+        this.setupAudioTrack(evt.track);
+      }
+    });
+    this.dailyCallObject.on("track-stopped", this.handleEventToConsole.bind(this));
+    this.dailyCallObject.on("participant-joined", this.handleEventToConsole.bind(this));
+    this.dailyCallObject.on("participant-updated", this.handleEventToConsole.bind(this));
+    this.dailyCallObject.on("participant-left", () => {
+      // When the bot leaves, we are also disconnecting from the call
+      this.disconnect()
+    });
+    this.dailyCallObject.on("left-meeting", () => {
+      this.updateStatus('Disconnected');
+      this.connectBtn.disabled = false;
+      this.disconnectBtn.disabled = true;
+      this.log('Client disconnected');
+    });
+    this.dailyCallObject.on("error", this.handleEventToConsole.bind(this));
+  }
+
+  /**
+   * Set up an audio track for playback
+   * Handles both initial setup and track updates
+   */
+  setupAudioTrack(track) {
+    this.log(`Setting up audio track, track state: ${track.readyState}, muted: ${track.muted}`);
+
+    // Check if we're already playing this track
+    if (this.botAudio.srcObject) {
+      const oldTrack = this.botAudio.srcObject.getAudioTracks()[0];
+      if (oldTrack?.id === track.id) return;
+    }
+    // Create a new MediaStream with the track and set it as the audio source
+    this.botAudio.srcObject = new MediaStream([track]);
+    this.botAudio.onplaying = async (event) => {
+      this.log("onplaying")
+      this.log("Will send the audio message to play the audio at the next tick")
+      this.dailyCallObject.sendAppMessage("playable")
```

Comment:
The "onplaying" event is supposed to trigger only after playback has started (spec here). While it helps, it does not completely solve the issue. 

To avoid any audio clipping, if we rely only on this approach, we would still need to add a timeout of at least 1 second before sending the app message.